### PR TITLE
Added SPADE module and relative sources and test files

### PR DESCRIPTION
**Pull request by pietroquaglio**
This pull request add the Spike PAttern Detection and Evaluation (SPADE) method [[1]](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3805944/), [[2]](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5443150/). This is a combination of mining and statistical test for detection of significant spike patterns in massively parallel spiketrains.

### CODE REVIEW

**Comment by alperyeg** [comment]
We need to supply a proper installation script so that travis works. Also a new testing environment is needed to test for mpi.

File: elephant/spade.py [hunk 1]
```diff
@@ -1001,18 +1062,30 @@ def approximate_stability(concepts, rel_matrix, n_subsets, delta=0, epsilon=0):
 
     Parameters
     ----------
-    concepts:
-
+    concepts: ???
```
Comment (mdenker) [inline elephant/spade.py]: These two parameters need a description...

File: elephant/spade.py [hunk 2]
```diff
@@ -1001,18 +1062,30 @@ def approximate_stability(concepts, rel_matrix, n_subsets, delta=0, epsilon=0):
...
-    If delta + epsilon == 0 then an optimal n_subset is calculated according to
-    the formula given in Babin, Kuznetsov (2012) (Proposition 6):
-     ..math::
-                n_subset = frac{1}{2\eps^2} \ln(frac{2}{\delta}) +1
+        epsilon: absolute error
+        Default: 0
+
+    Output
+    ------
+    stability: list
```
Comment (mdenker) [inline elephant/spade.py]: Can you please specify the output a bit more in detail?

File: elephant/spade.py (hunk 1 above)
Comment (pietroquaglio) [reply elephant/spade.py]: Addressed in the following commit (858c363)

File: elephant/spade.py (hunk 2 above)
Comment (pietroquaglio) [reply elephant/spade.py]: Addressed in the following commit (858c363). Thanks for comments!

**Comment by mdenker** [comment]
Failing travis on 'ubuntu' environment seems unrelated to this PR. From my side, the SPADE function could be merged.

**Pull request merged**